processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryConfig.java

```diff
@@ -48,6 +48,8 @@ public class GroupByQueryConfig
   private static final String CTX_KEY_BUFFER_GROUPER_INITIAL_BUCKETS = "bufferGrouperInitialBuckets";
   private static final String CTX_KEY_BUFFER_GROUPER_MAX_LOAD_FACTOR = "bufferGrouperMaxLoadFactor";
   private static final String CTX_KEY_MAX_ON_DISK_STORAGE = "maxOnDiskStorage";
+  private static final String CTX_KEY_MAX_SELECTOR_DICTIONARY_SIZE = "maxSelectorDictionarySize";
+  private static final String CTX_KEY_MAX_MERGING_DICTIONARY_SIZE = "maxMergingDictionarySize";
```

Comment:
We should mention that this is configurable per query in the java docs as well.

Reply:
I've added it to the docs [here](https://github.com/apache/druid/pull/15354/commits/d8db3b5974b4da539761f47de3683a65e87c5d78#diff-4b1a32cb60c43ce394b2898a895c7050c987788b81b2461fb9c8f971efec8040R381) 

any other places I should add it?

Reply:
Kinda weird that [docs/configuration/index.md](https://github.com/apache/druid/blob/67c7b6248cfead7b9ab6950e13cc33583dbf8b88/docs/configuration/index.md?plain=1#L2164) says that we can override, even though overriding was a no-op.

Looking up the blame, it is mentioned here https://github.com/apache/druid/pull/12763, that removing it was intentional. Perhaps we should not add it back then, and fix up the discrepancy above WDYT?

Reply:
It was removed from the context as setting it might not be a common case + "complicates calculation" 
...however as the tests also rely on similar ways...I went to add it back
I believe its better to have all options follow the same convention... instead of doing something different just for these options...(which have silently caused that this feature become untested)

fyi: @gianm

Reply:
Can you also modify https://github.com/apache/druid/blob/67c7b6248cfead7b9ab6950e13cc33583dbf8b88/docs/configuration/index.md?plain=1#L2164 as the part of the fix here to mention that it is not just lower, but override. Someone can intentionally or accidentally set it to a higher value.